src/pages/library/dataBrowser-utils.js

```diff
@@ -123,14 +132,34 @@ export const useDataCatalog = () => {
   return { dataCatalog, refresh, loading }
 }
 
-export const importDataToWorkspace = datasets => {
-  // TODO (DC-284): Call data catalog to figure out what the format should be for importing to workspace
-  const format = 'snapshot'
-  Nav.history.push({
-    pathname: Nav.getPath('import-data'),
-    search: qs.stringify({
-      url: getConfig().dataRepoUrlRoot, format, referrer: 'data-catalog',
-      snapshotIds: _.map('dct:identifier', datasets)
-    })
-  })
+export const importDataToWorkspace = async (dataset, asyncHandler) => {
+  const routeOptions = await Utils.cond(
+    [isWorkspace(dataset), () => {
+      return {
+        pathname: Nav.getPath('import-data'),
+        search: qs.stringify({
+          format: 'catalog',
+          snapshotName: dataset['dct:title'],
+          catalogDatasetId: dataset.id
+        })
+      }
+    }],
+    [
+      isDatarepoSnapshot(dataset), async () => {
+        asyncHandler()
+        const jobInfo = await Ajax().DataRepo.snapshot(dataset['dct:identifier']).exportSnapshot()
+        await poll(async () => await Ajax().DataRepo.job(jobInfo.id).details(), 1000, jobStatus => jobStatus['job_status'] !== 'running')
+        const jobResult = await Ajax().DataRepo.job(jobInfo.id).result()
+        const jobResultManifest = jobResult && jobResult.format && jobResult.format.parquet && jobResult.format.parquet.manifest
```

Comment:
can you instead use `?.`?
```suggestion
        const jobResultManifest = jobResult?.format?.parquet?.manifest
```